radar/radar.go

```diff
@@ -108,47 +125,48 @@ func (radar *Radar) Scan(logger lager.Logger, resourceName string) error {
 	return radar.scan(logger, resourceName)
 }
 
-func (radar *Radar) scan(logger lager.Logger, resourceName string) error {
+func (radar *Radar) scan(logger lager.Logger, resourceName string) (err error) {
 	pipelinePaused, err := radar.db.IsPaused()
 	if err != nil {
 		logger.Error("failed-to-check-if-pipeline-paused", err)
-		return err
+		return
 	}
 
 	if pipelinePaused {
 		logger.Debug("pipeline-paused")
-		return nil
+		return
 	}
 
-	config, _, err := radar.db.GetConfig()
+	savedResource, err := radar.db.GetResource(resourceName)
 	if err != nil {
-		logger.Error("failed-to-get-config", err)
-		// don't propagate error; we can just retry next tick
-		return nil
+		return
 	}
 
-	resourceConfig, found := config.Resources.Lookup(resourceName)
-	if !found {
-		logger.Info("resource-removed-from-configuration")
-		// return an error so that we exit
-		return resourceNotConfiguredError{ResourceName: resourceName}
+	if savedResource.Paused {
+		return
 	}
 
-	savedResource, err := radar.db.GetResource(resourceName)
+	resourceConfig, err := radar.getResourceConfig(logger, resourceName)
 	if err != nil {
-		return err
+		return
 	}
 
-	if savedResource.Paused {
-		return nil
-	}
+	defer func() {
+		if resourceConfig.CheckEvery != "" {
+			radar.interval, err = time.ParseDuration(resourceConfig.CheckEvery)
```

Comment:
@vito - yup you totally nailed it and that was the thing I would worried about (that it would be tricky to parse out).  So it's named returns all the way down now.  Since this code has to be deferred there isn't a really good way to get values out of any function that is deferred unless the variables are pre-declared in the outer scope.

I am thinking this may be a bit of necessary complexity?  I could also add a comment to point out the fact that we've gone to named returns.